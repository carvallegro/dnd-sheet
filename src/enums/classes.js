const classes = {
  BARBARIAN: 'BARBARIAN',
  BARD: 'BARD',
  CLERIC: 'CLERIC',
  DRUID: 'DRUID',
  FIGHTER: 'FIGHTER',
  MONK: 'MONK',
  PALADIN: 'PALADIN',
  RANGER: 'RANGER',
  ROGUE: 'ROGUE',
  SORCERER: 'SORCERER',
  WARLOCK: 'WARLOCK',
  WIZARD: 'WIZARD'
}

export default classes
