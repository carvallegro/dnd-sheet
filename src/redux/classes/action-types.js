export const SELECT_CLASS = 'SELECT_CLASS'
