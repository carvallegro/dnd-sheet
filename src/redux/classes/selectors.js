import _ from 'lodash'

export const selectSelectedClass = state =>
  _(state).get('classes.selectedClass', {})
export const selectSelectedClassIndex = state =>
  _(state).get('classes.selectedClass.index')
