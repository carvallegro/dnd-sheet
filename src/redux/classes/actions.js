import { createAction } from 'redux-actions'
import { SELECT_CLASS } from '@redux/classes/action-types'

export const selectClass = createAction(SELECT_CLASS)
