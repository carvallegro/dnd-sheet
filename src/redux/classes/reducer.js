import { handleActions } from 'redux-actions'
import { selectClass } from '@redux/classes/actions'

const INITIAL_STATE = {
  classes: {
    selectedClass: {
      index: null
    }
  }
}

const classesReducer = handleActions(
  {
    [selectClass]: (state, action) => ({
      ...state,
      selectClass: {
        index: action.payload
      }
    })
  },
  INITIAL_STATE
)

export default classesReducer
