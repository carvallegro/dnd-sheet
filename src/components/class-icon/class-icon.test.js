import _ from 'lodash'
import React from 'react'
import { shallow } from 'enzyme'

import classes from '../../enums/classes'

import ClassIcon from './index'

describe('<ClassIcon />', () => {
  it('should render', () => {
    const wrapper = shallow(<ClassIcon />)
    expect(wrapper).toMatchSnapshot()
  })

  _(classes)
    .values()
    .forEach(c =>
      it(`should render ${c} icon`, () => {
        const wrapper = shallow(<ClassIcon classEnum={c} />)
        expect(wrapper).toMatchSnapshot()
      })
    )
})
