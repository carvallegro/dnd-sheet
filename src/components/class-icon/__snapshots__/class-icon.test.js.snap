// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassIcon /> should render 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render BARBARIAN icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render BARD icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render CLERIC icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render DRUID icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render FIGHTER icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render MONK icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render PALADIN icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render RANGER icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render ROGUE icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render SORCERER icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render WARLOCK icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;

exports[`<ClassIcon /> should render WIZARD icon 1`] = `
<styled.div
  height="100%"
>
  <ReactComponent />
</styled.div>
`;
